### ISSUE TITLE
Support for CSP / Delegated Administration

### ISSUE THREAD

**Issue by mtwelve** [microsoftgraph/powershell-intune-samples#26]
Is it possible to tweak the scripts or add support for access Intune tenants managed through CSP / Delegated Administration?

**Comment by davefalkus** [comment]
Hi @mtwelve 

Could you provide more details on your request? To set delegated admin by a Global Admin account there is the following - https://github.com/microsoftgraph/powershell-intune-samples/tree/master/AdminConsent

From there anyone without GA rights can run Graph calls against Intune if they have permissions to complete actions.

Thanks

Dave

**Comment by davefalkus** [comment]
Hi @mtwelve,

Does the above help with your question?

Closing the issue, please re-open if the suggestion doesn't help.

Thanks

Dave

**Comment by mtwelve** [comment]
Apologies for the delay

It's not so much about granting the permission, it is about being able to access the Intune tenant of a customer using our credentials.
For example we are engineer@cspprovider.com and our customers have customer1.onmicrosoft.com and customer2.onmicrosoft.com.

When you run one of the scripts you are asked to authenticate by providing a User Principal Name.
I can authenticate fine but then I am connected to our @cspprovider.com Intune tenant.

How can I instead authenticate with engineer@cspprovider.com but connect to the customer1.onmicrosoft.com tenant?

We administer the Intune tenant using the Azure Portal GUI using our engineer@cspprovider.com credentials.

**Comment by davefalkus** [comment]
Hi @mtwelve 

Have you reviewed the following documentation? It talks about creating an application available to other tenants for authentication.

https://docs.microsoft.com/en-us/intune/intune-graph-apis 

Let me know if you require further information.

Dave

**Comment by mtwelve** [comment]
Hello,

That has helped. Here are the changes that I had to make in order to get it to work. It may be possible to roll these in the scripts to support this scenario.

Rather than the authority url be derived from what the user enters as the user principal name I manually changed this to a specific customer (for testing)

`$authority = "https://login.microsoftonline.com/$Tenant"`
to
`$authority = "https://login.microsoftonline.com/customer1.onmicrosoft.com"`

Despite us having Global Administrator rights on the Azure Subscription, Azure AD and Intune delegated to us through them being a managed customer on our CSP Portal I was unable to run the _GA_AdminConsent_Set.ps1_ and it prompt me to accept the rights assignment for the Enterprise Application/Service Principal. I had to first:

- Invite my UPN as a guest account in the customer AAD
- Assign the Global Admin role to my new 'guest' account
- Re-run the script

I can then lower the role assignment to Intune Service Administrator and it appears to still run.

It would seem that two changes would make this work:
- Asking the user when running the script if they are connecting a different tenant, accepting the input and updating the authority url
- Somehow create the Service Principal using the delegated authentication though I don't know how you would then assign access to the new Service Principal as our accounts or a group don't appear in the customers AAD.. despite us having the rights listed at the start of this post

Does this all make sense?
